### PR TITLE
First Commit

### PR DESCRIPTION
**Pull request by gaurav-unifydots**

### CODE REVIEW

**Review by vivek-unifydots — CHANGES_REQUESTED** [review]
@gaurav-unifydots please address the comments

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply .idea/.gitignore:1]: done

File: lib/jcommander-1.48.jar
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply lib/jcommander-1.48.jar:1]: done

[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply lib/jcommander-1.48.jar:1]: done

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+			<version>3.11.0</version>
+		</dependency>
+
+		<dependency>
+			<groupId>org.seleniumhq.selenium</groupId>
+			<artifactId>selenium-firefox-driver</artifactId>
+			<version>3.11.0</version>
+		</dependency>
+
+		<dependency>
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply pom.xml:87]: done

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+			<maven.compiler.source>17</maven.compiler.source>
+			<maven.compiler.target>17</maven.compiler.target>
+		<testng.files.location>src/test/resources/</testng.files.location>
+		<testng.file>testng.xml</testng.file>
+	</properties>
+
+
+	<repositories>
+		<repository>
+			<id>maven.oracle.com</id>
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply pom.xml:25]: removed

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+		<dependency>
+			<groupId>commons-dbutils</groupId>
+			<artifactId>commons-dbutils</artifactId>
+			<version>1.5</version>
+		</dependency>
+		
+		
+		
+		
+
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply pom.xml:128]: removed

File: src/test/java/com/unifydots/tests/FirstTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    public   void setUp1(){
+
+
+    }
+    @Test
+    public void PrintNumbers() throws InterruptedException {
+Thread.sleep(5000);
+driver.findElement(By.xpath("//*[@type='email']")).sendKeys("gaurav");
+        System.out.println("i"+2);
+        //Assert.assertFalse(true);
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/FirstTest.java:36]: done

File: src/test/java/com/unifydots/tests/FirstTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    }
+
+    @AfterMethod
+    public  void shutDown(){
+
+        driver.close();
+    }
+    /*@AfterSuite
+    public void afterSuite(){ //executed after all the tests have ran
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/FirstTest.java:46]: done

File: src/test/resources/testng.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="Component Tests">
+    <!--<listeners>
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply src/test/resources/testng.xml:5]: done

File: src/test/java/com/unifydots/tests/SecondTest.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        setDesiredBrowser("chrome");
+        this.driver=BaseTest.getDriver();
+
+    }
+
+    @Test
+    public void PrintNumbers(){
+
+        System.out.println("SecondTest : "+2);
+        //Assert.assertFalse(true);
```
[the remark this answers is not in the document]
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/SecondTest.java:26]: done

File: .gitignore.bak [hunk 9]
```diff
@@ -0,0 +1,7 @@
+target/
```
Comment (vivek-unifydots) [inline .gitignore.bak:1]: please remove this file

File: .idea/workspace.xml
Comment (vivek-unifydots) [inline .idea/workspace.xml:1]: please remove .idea folder from commit

File: allure-results/03472a10-e11f-48c5-acf8-680498cce8d9-container.json
Comment (vivek-unifydots) [inline allure-results/03472a10-e11f-48c5-acf8-680498cce8d9-container.json:1]: please do not include results into commit

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+            <version>3.11.0</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-firefox-driver</artifactId>
+            <version>3.11.0</version>
+        </dependency>
+
+        <dependency>
```
Comment (vivek-unifydots) [inline pom.xml:63]: please remove unused dependencies

File: src/test/java/com/unifydots/tests/FirstTest.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class FirstTest extends BaseTest {
+    public static WebDriver driver;
+    public ExtentReports extent = new ExtentReports();
```
Comment (vivek-unifydots) [inline src/test/java/com/unifydots/tests/FirstTest.java:16]: please remove extent report code

File: src/test/java/com/unifydots/tests/SecondTest.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+public class SecondTest extends BaseTest {
+    public static WebDriver driver;
+
+    @Test(description = "Second Test")
+    public void PrintNumbers(){
+
+        System.out.println("SecondTest : "+2);
+        //Assert.assertFalse(true);
+      //  this.driver=BaseUtil.driver;
+        driver.findElement(By.xpath("//[@email='email']")).sendKeys("gaurav");
```
Comment (vivek-unifydots) [inline src/test/java/com/unifydots/tests/SecondTest.java:22]: locators should be part of page classes

File: .gitignore.bak (hunk 9 above)
Comment (gaurav-unifydots) [reply .gitignore.bak:1]: done

File: .idea/workspace.xml
Comment (gaurav-unifydots) [reply .idea/workspace.xml:1]: done

File: allure-results/03472a10-e11f-48c5-acf8-680498cce8d9-container.json
Comment (gaurav-unifydots) [reply allure-results/03472a10-e11f-48c5-acf8-680498cce8d9-container.json:1]: done

File: pom.xml (hunk 10 above)
Comment (gaurav-unifydots) [reply pom.xml:63]: done

File: src/test/java/com/unifydots/tests/FirstTest.java (hunk 11 above)
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/FirstTest.java:16]: done

File: src/test/java/com/unifydots/tests/SecondTest.java (hunk 12 above)
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/SecondTest.java:22]: this was for testing purpose.I will make page class and move it there

File: src/main/java/com/unifydots/pages/LoginPage.java [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+WebDriver driver;
+
+    public LoginPage(WebDriver driver) {
+
+        this.driver=driver;
+        PageFactory.initElements(driver,this);
+    }
+
+
+    public void login() throws InterruptedException {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/pages/LoginPage.java:19]: https://opensource-demo.orangehrmlive.com/web/index.php/auth/login

Comment (vivek-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:19]: pass username and password as parameters

File: src/test/java/com/unifydots/tests/FirstTest.java [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class FirstTest extends BaseTest {
+
+    @Test(description = "First Test")
+    public void PrintNumbers() throws InterruptedException {
+loginPage.login();
```
Comment (vivek-unifydots) [inline src/test/java/com/unifydots/tests/FirstTest.java:19]: fix formatting

File: src/main/java/com/unifydots/pages/LoginPage.java [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+        Thread.sleep(5000);
+        driver.findElement(By.xpath("//*[@type='email']")).sendKeys("gaurav");
+
+    }
+
+    public void login1(){
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/pages/LoginPage.java:25]: please remove

File: src/main/java/com/unifydots/pages/LoginPage.java [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+        System.out.println("SecondTest : "+2);
+        //Assert.assertFalse(true);
+        //  this.driver=BaseUtil.driver;
+        driver.findElement(By.xpath("//[@email='email']")).sendKeys("gaurav");
+    }
+
+    public  void shutDown(){
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/pages/LoginPage.java:33]: @AfterTest or @AfterClass

File: src/main/java/com/unifydots/utility/BaseTest.java [hunk 17]
```diff
@@ -0,0 +1,596 @@
+/**
+ *
+ */
+package com.unifydots.utility;
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/utility/BaseTest.java:4]: java/com/unifydots/tests/base

File: src/main/java/com/unifydots/utility/BaseTest.java [hunk 18]
```diff
@@ -0,0 +1,583 @@
...
+	 */
+	public FileInputStream commaonConfigLoc;
+	/**
+	 * Variable for applicationUrl.
+	 */
+	private static String applicationUrl;
+	/**
+	 * Initialization of EventFiringWebDriver class
+	 */
+	public static EventFiringWebDriver e_driver;
```
[the remark this answers is not in the document]
Comment (vivek-unifydots) [reply src/main/java/com/unifydots/utility/BaseTest.java:87]: please remove

File: src/test/java/com/unifydots/tests/SecondTest.java [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+public class SecondTest extends BaseTest {
+
+    @Test(description = "Second Test")
+    public void PrintNumbers(){
+
+        loginPage.login1();
+loginPage.shutDown();
```
Comment (vivek-unifydots) [inline src/test/java/com/unifydots/tests/SecondTest.java:19]: please fix formatting

File: src/test/resources/testng.xml [hunk 20]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="Component Tests">
+
+    <test name="Unify Dots Tests" preserve-order="false" parallel="false" thread-count="2">
```
Comment (vivek-unifydots) [inline src/test/resources/testng.xml:6]: fix parallel

File: src/test/java/com/unifydots/tests/FirstTest.java [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+loginPage.shutDown();
+
+
+    }
+
```
Comment (vivek-unifydots) [inline src/test/java/com/unifydots/tests/FirstTest.java:24]: remove extra spaces

File: src/test/java/com/unifydots/tests/FirstTest.java (hunk 14 above)
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/FirstTest.java:19]: done

File: src/main/java/com/unifydots/pages/LoginPage.java (hunk 16 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:33]: done

File: src/main/java/com/unifydots/pages/LoginPage.java (hunk 13 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:19]: done

File: src/test/java/com/unifydots/tests/FirstTest.java (hunk 14 above)
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/FirstTest.java:19]: done

File: src/main/java/com/unifydots/pages/LoginPage.java (hunk 15 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:25]: done

Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:25]: done

File: src/main/java/com/unifydots/pages/LoginPage.java (hunk 16 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:33]: done

File: src/main/java/com/unifydots/utility/BaseTest.java (hunk 17 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/utility/BaseTest.java:4]: done

File: src/main/java/com/unifydots/pages/LoginPage.java (hunk 16 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/pages/LoginPage.java:33]: done

File: src/main/java/com/unifydots/utility/BaseTest.java (hunk 17 above)
Comment (gaurav-unifydots) [reply src/main/java/com/unifydots/utility/BaseTest.java:4]: done

File: src/test/java/com/unifydots/tests/SecondTest.java (hunk 19 above)
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/SecondTest.java:19]: done

File: src/test/resources/testng.xml (hunk 20 above)
Comment (gaurav-unifydots) [reply src/test/resources/testng.xml:6]: done

File: src/test/java/com/unifydots/tests/FirstTest.java (hunk 21 above)
Comment (gaurav-unifydots) [reply src/test/java/com/unifydots/tests/FirstTest.java:24]: done

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 22]
```diff
@@ -0,0 +1,622 @@
...
+
+    /**
+     * FileInputStream class reference variable for common config file location
+     */
+    public FileInputStream commaonConfigLoc;
+    /**
+     * Variable for applicationUrl.
+     */
+    private static String applicationUrl;
+    /**
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:80]: please remove

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 23]
```diff
@@ -0,0 +1,622 @@
...
+
+            // Initialize "Chrome" as default browser
+            default:
+                WebDriverManager.chromedriver().setup();
+                driver = new ChromeDriver();
+                break;
+        }
+        /**
+         * uncomment below mentioned code to see line by line execution description
+         * e_driver = new EventFiringWebDriver(driver); // Now create object of
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:160]: please remove all this

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 24]
```diff
@@ -0,0 +1,622 @@
...
+        driver.manage().timeouts().implicitlyWait(SeleniumConstant.IMPLICIT_WAIT, TimeUnit.SECONDS);
+        driver.manage().timeouts().setScriptTimeout(SeleniumConstant.SETSCRIPT_TIMEOUT, TimeUnit.SECONDS);
+        driver.get("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login");
+        return driver;
+    }
+
+    /**
+     * Method to get the title of current page
+     */
+    public String getTitleOfApplication() {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:176]: getTitle()

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 25]
```diff
@@ -0,0 +1,622 @@
...
+        return driver.getTitle();
+    }
+
+    /**
+     * Method to get the current url of the application
+     */
+    public String getCurrentOfApplication() {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:183]: getCurrentUrl()

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 26]
```diff
@@ -0,0 +1,622 @@
...
+
+    /**
+     * This method is used to get a particular date in a particular format.
+     *
+     * @param offsetFromToday Number of days to be added or subtracted from today.
+     * @param format          Required format of the date example dd.MM.yyyy
+     * @return date in String format
+     */
+    @SuppressWarnings("static-method")
+    public String getDateInFormattedString(int offsetFromToday, String format) {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:273]: please move this to a new class DateUtils

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 27]
```diff
@@ -0,0 +1,622 @@
...
+    /**
+     * This method is used to get a particular date in a particular format.
+     *
+     * @param offsetFromTodayInMonths Number of months to be added or subtracted
+     *                                from today.
+     * @param format                  Required format of the date example dd.MM.yyyy
+     * @return date in String format
+     */
+    @SuppressWarnings("static-method")
+    public String getDateInFormattedStringMonths(int offsetFromTodayInMonths, String format) {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:291]: please move this to a new class DateUtils

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 28]
```diff
@@ -0,0 +1,622 @@
...
+
+    /**
+     * Method which returns the date object from the String date passed. String date
+     * has to be in the format of 'hours:minutes:seconds'
+     *
+     * @param date String date which has to be converted
+     * @return formatted date object; used internally
+     */
+    @SuppressWarnings("static-method")
+    private Date getCurrentDateFromTable(String date) {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:366]: from which table this method is getting date

File: unifydots.qa.iml
Comment (vivek-unifydots) [inline unifydots.qa.iml:1]: is this file required to be committed?

File: src/main/java/com/unifydots/base/BaseTest.java [hunk 29]
```diff
@@ -0,0 +1,622 @@
...
+
+            public WebElement apply(WebDriver input) {
+
+                return driver.findElement(locator);
+            }
+        });
+        return waitElement;
+    }
+
+    public static void takeScreenshotAtEndOfTest() throws IOException {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/base/BaseTest.java:581]: from where this method is being called.. this should be part of listener

File: src/main/java/com/unifydots/utility/DateUtils.java [hunk 30]
```diff
@@ -0,0 +1,80 @@
...
+        return dateObj;
+    }
+
+
+    /**
+     * Used internally to format system date as per the requirement.
+     *
+     * @return formatted date as per the requirement
+     */
+    protected Date dateFormatConversion() {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/utility/DateUtils.java:72]: getCurrentDate()

File: src/main/java/com/unifydots/utility/DateUtils.java [hunk 31]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * Method which returns the date object from the String date passed. String date
+     * has to be in the format of 'hours:minutes:seconds'
+     *
+     * @param date String date which has to be converted
+     * @return formatted date object; used internally
+     */
+    @SuppressWarnings("static-method")
+    private Date getCurrentDateFormat(String date) {
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/utility/DateUtils.java:54]: please remove

File: src/test/java/com/unifydots/tests/PropertiesTest.java [hunk 32]
```diff
@@ -0,0 +1,19 @@
...
+import java.io.IOException;
+import java.util.Properties;
+
+public class PropertiesTest  {
+
+
+    @Test(description = "Properties Test")
+    public void PrintNumbers() throws IOException {
+        Properties application=BaseTest.readPropertiesFileContents("EN","COMMON");
+        System.out.println("base url value from property file "+application.getProperty("base.url"));
```
Comment (vivek-unifydots) [inline src/test/java/com/unifydots/tests/PropertiesTest.java:15]: Log.info

File: unifydots.qa.iml
Comment (vivek-unifydots) [inline unifydots.qa.iml:1]: please remove

File: src/main/java/com/unifydots/pages/LoginPage.java [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
+    @Severity(SeverityLevel.BLOCKER)
+    @Step("Login to the Application")
+    public void login(String UserName, String PassWord) {
+        userName.sendKeys(UserName);
+        password.sendKeys(PassWord);
+        loginButton.click();
+
+    }
+
+    @Step
```
Comment (vivek-unifydots) [inline src/main/java/com/unifydots/pages/LoginPage.java:45]: please add description

File: pom.xml [hunk 34]
```diff
@@ -0,0 +1,181 @@
...
+                <version>2.20</version>
+                <configuration>
+                    <argLine>
+                        -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
+                    </argLine>
+                    <testFailureIgnore>true</testFailureIgnore>
+
+                    <suiteXmlFiles>
+                        <!--
+                                                <classpathScope>test</classpathScope>
```
Comment (vivek-unifydots) [inline pom.xml:153]: please format or remove

**Pull request merged**